Mathlib/AlgebraicGeometry/ProjectiveSpectrum/Scheme.lean

```diff
@@ -155,164 +155,67 @@ For any `x` in `Proj| (pbo f)`, the corresponding ideal in `Spec A⁰_f`. This f
-    ∃ (c : algebraMap A (Away f) '' x.1.asHomogeneousIdeal →₀ Away f) (N : ℕ) (acd :
-      ∀ y ∈ c.support.image c, A),
-      f ^ N • z.val =
-        algebraMap A (Away f)
-          (∑ i in c.support.attach,
-            acd (c i) (Finset.mem_image.mpr ⟨i, ⟨i.2, rfl⟩⟩) * i.1.2.choose) :=
-  MemCarrier.clear_denominator' x <| (mem_carrier_iff x z).mpr hz
-#align algebraic_geometry.Proj_iso_Spec_Top_component.to_Spec.mem_carrier.clear_denominator AlgebraicGeometry.ProjIsoSpecTopComponent.ToSpec.MemCarrier.clear_denominator
-
-/-- For `x : pbo f`, the ideal `A⁰_f ∩ span {g / 1 | g ∈ x}`
-is equal to `span {a/f^n | a ∈ x and deg(a) = deg(f^n)}`. -/
-lemma carrier_eq_span :
-    carrier x =
-    Ideal.span { z : HomogeneousLocalization.Away 𝒜 f |
-      ∃ (s F : A) (_ : s ∈ x.1.asHomogeneousIdeal) (n : ℕ)
-        (s_mem : s ∈ 𝒜 n) (F_mem1 : F ∈ 𝒜 n) (F_mem2 : F ∈ Submonoid.powers f),
-        z = Quotient.mk'' ⟨n, ⟨s, s_mem⟩, ⟨F, F_mem1⟩,F_mem2⟩ } := by
-  classical
-  refine le_antisymm ?_ <| Ideal.span_le.mpr ?_
-  · intro z hz
-    let k : ℕ := z.den_mem.choose
-    have hk : f^k = z.den := z.den_mem.choose_spec
-    suffices mem1 : z.num ∈ x.1.asHomogeneousIdeal by
-      refine Ideal.subset_span ⟨_, _, mem1, _, z.num_mem_deg, z.den_mem_deg, z.den_mem, ?_⟩
-      rw [HomogeneousLocalization.ext_iff_val, HomogeneousLocalization.val_mk'',
-        HomogeneousLocalization.eq_num_div_den]
-
-    obtain ⟨c, N, acd, eq1⟩ := MemCarrier.clear_denominator x hz
-    rw [z.eq_num_div_den, smul_mk, smul_eq_mul, ← mk_one_eq_algebraMap,
-      Localization.mk_eq_mk_iff, Localization.r_iff_exists] at eq1
-    obtain ⟨⟨_, ⟨l, rfl⟩⟩, eq1⟩ := eq1
-    dsimp only [OneMemClass.coe_one] at eq1
-    rw [one_mul, ← hk, ← mul_assoc, ← mul_assoc, ← pow_add, ← pow_add] at eq1
-
-    suffices f^(l + k) * ∑ i in c.support.attach, acd (c i) _ * i.1.2.choose ∈
-      x.1.asHomogeneousIdeal from
-      (x.1.isPrime.mem_or_mem (eq1.symm ▸ this)).resolve_left fun r ↦
-        (ProjectiveSpectrum.mem_basicOpen 𝒜 _ _).mp x.2 <| x.1.isPrime.mem_of_pow_mem _ r
-    exact Ideal.mul_mem_left _ _ <| Ideal.sum_mem _ (fun ⟨j, hj⟩ _ ↦
-      Ideal.mul_mem_left _ _ j.2.choose_spec.1)
-
-  · rintro _ ⟨s, _, hs, n, s_mem, F_mem, ⟨l, rfl⟩, rfl⟩
-    erw [mem_carrier_iff, HomogeneousLocalization.val_mk'']
-    erw [show (Localization.mk s ⟨f ^ l, ⟨_, rfl⟩⟩ : Localization.Away f) =
-      (Localization.mk 1 ⟨f^l, ⟨_, rfl⟩⟩ : Localization.Away f) * Localization.mk s 1 by
-      rw [Localization.mk_mul, mul_one, one_mul]]
-    exact Ideal.mul_mem_left _ _ <| Ideal.subset_span ⟨s, hs, rfl⟩
+theorem mem_carrier_iff_exists (z : A⁰_ f) :
+    z ∈ carrier x ↔ ∃ y ∈ x.1.asHomogeneousIdeal, ∃ n, f ^ n • z.val = algebraMap _ _ y := by
+  rw [carrier, Ideal.mem_comap, isLocalization.mem_map_algebraMap_iff]
+  simp [Submonoid.mem_powers_iff, mul_comm z.val, Algebra.smul_def]
+
+theorem MemCarrier.eventually_mem_ideal {z : A⁰_ f} (hz : z ∈ carrier x) :
```

Comment:
Maybe `eventually_mem_ideal_of_mem_carrier` instead, since the namespace does not enable dot notation.

Reply:
I think both are fine